src/libraries/Common/src/SourceGenerators/TypeModelHelper.cs

```diff
@@ -32,5 +34,62 @@ void TraverseContainingTypes(INamedTypeSymbol current)
                 }
             }
         }
+
+        public static string GetFullyQualifiedName(this ITypeSymbol type) => type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+
+        /// <summary>
+        /// Removes any type metadata that is erased at compile time, such as NRT annotations and tuple labels.
+        /// </summary>
+        public static ITypeSymbol EraseCompileTimeMetadata(this Compilation compilation, ITypeSymbol type)
```

Comment:
@eiriktsarpalis why's this necessary? The `ITypeSymbol`'s don't end up in the incremental models.

Reply:
In the case of STJ we want to strip any nullability information for metadata that is being generated. For example, we want to generate a `JsonTypeInfo<List<string>>` even if the user is specifying a `JsonSerializable(typeof(List<string?>))`. CB doesn't use a runtime metadata model so it may or may not be applicable.

Reply:
I don't believe it is; I might revert this back to STJ only & add a test.